internal/builtinfunctions/functions.go

```diff
@@ -0,0 +1,428 @@
+			schema.PointerTo(
+				"Converts an integer to a string whose characters represent that integer in base-10.\n"+
+					"For example, an input of `55` will output \"55\"",
+			),
+			nil,
+		),
+		func(a int64) string {
+			return strconv.FormatInt(a, 10)
+		},
+	)
+	if err != nil {
+		panic(err)
+	}
+	return funcSchema
+}
+
+func getFloatToStringFunction() schema.CallableFunction {
+	funcSchema, err := schema.NewCallableFunction(
+		"floatToString",
+		[]schema.Type{schema.NewFloatSchema(nil, nil, nil)},
+		schema.NewStringSchema(nil, nil, regexp.MustCompile(`^\d+\.\d*$`)),
+		false,
+		schema.NewDisplayValue(
+			schema.PointerTo("floatToString"),
+			schema.PointerTo(
+				"Converts a floating point number to a string whose characters"+
+					"represent that number in base-10 as as simple decimal.\n"+
+					"For example, an input of `5000.5` will output \"5000.5\"",
+			),
+			nil,
+		),
+		func(a float64) string {
+			return strconv.FormatFloat(a, 'f', -1, 64)
+		},
+	)
+	if err != nil {
+		panic(err)
+	}
+	return funcSchema
+}
+
+// TODO: Webb can add a function called floatToStringAdvanced that allows other float formats
+
+func getBooleanToStringFunction() schema.CallableFunction {
+	funcSchema, err := schema.NewCallableFunction(
+		"boolToString",
+		[]schema.Type{schema.NewBoolSchema()},
+		schema.NewStringSchema(nil, nil, regexp.MustCompile(`^true|false$`)),
+		false,
+		schema.NewDisplayValue(
+			schema.PointerTo("boolToString"),
+			schema.PointerTo(
+				"Represents `true` as \"true\", and `false` as \"false\".",
+			),
+			nil,
+		),
+		strconv.FormatBool, // Wrap go standard lib function.
+	)
+	if err != nil {
+		panic(err)
+	}
+	return funcSchema
+}
+
+func getStringToIntFunction() schema.CallableFunction {
+	funcSchema, err := schema.NewCallableFunction(
+		"stringToInt",
+		[]schema.Type{schema.NewStringSchema(nil, nil, regexp.MustCompile(`^-?(?:0|[1-9]\d*)$`))},
+		schema.NewIntSchema(nil, nil, nil),
+		true,
+		schema.NewDisplayValue(
+			schema.PointerTo("stringToInt"),
+			schema.PointerTo(
+				"Interprets the string as a base-10 integer. Can fail if the input is not a valid integer.",
+			),
+			nil,
+		),
+		func(s string) (int64, error) {
+			return strconv.ParseInt(s, 10, 0)
+		},
+	)
+	if err != nil {
+		panic(err)
+	}
+	return funcSchema
+}
+
+func getStringToFloatFunction() schema.CallableFunction {
+	funcSchema, err := schema.NewCallableFunction(
+		"stringToFloat",
+		[]schema.Type{schema.NewStringSchema(nil, nil, nil)},
+		schema.NewFloatSchema(nil, nil, nil),
+		true,
+		schema.NewDisplayValue(
+			schema.PointerTo("stringToFloat"),
+			schema.PointerTo(
+				"Converts the string s to a 64-bit floating-point number\n\n"+
+					"Accepts decimal and hexadecimal floating-point numbers\n"+
+					"as defined by the Go syntax for floating point literals\n"+
+					"https://go.dev/ref/spec#Floating-point_literals.\n"+
+					"If s is well-formed and near a valid floating-point number,\n"+
```

Comment:
The error text seen by the user is important; the identifier which the Go Standard Library uses to refer to it is not, unless the user is programming in Go.

The fact that the behavior of this function is different from the expression language seems like a cause for concern.  I think they should be consistent with each other...otherwise, someone is going to be surprised at some point.  (I don't care whether we adopt the more or the less restrictive of the two, but it's weird that the function would accept a value which we cannot express as a literal.)

Reply:
Well, they are values that we can express as a literal, just not in the form presented. So if someone inputs a hex number using stringToFloat, that's a valid value that will be stored in the 64 bit float. It just happens to be that we don't accept that format of the same value in the expression language.

The reason the stringToX functions should accept values that we do not is that the inputs into those functions may be from other tools or plugins. And you have extra context because you know which function you're calling, rather than needing to make sure the tokens make sense in the expression language.